lazy.md

```diff
@@ -13,3 +13,617 @@ author:
+    scheduler is used there is the possibility of stack overflow.
+- It is possible to `co_await just_error(e)` and `co_await just_stopped()`,
+    i.e., the sender isn't required to have a `set_value_t` completion.
+
+The `exec::task<T, C>` also provides a _context_ `C`. An object of
+this type becomes the environment for receivers `connect()`ed to
+`co_await`ed senders. The default context provides access to the
+task's scheduler. In addition an `in_place_stop_token` is provides
+which forwards the stop requests from the environment of the receiver
+which is connected to the task.
+
+Like the unifex task `exec::task<T, C>` doesn't provide any allocator
+support. When creating a task there are two allocations and an
+additional allocation for each `co_await`.
+
+# Objectives
+
+Also see [sender/receiver issue 241](https://github.com/cplusplus/sender-receiver/issues/241).
+
+Based on the prior work and discussions around corresponding coroutine
+support there a number of required or desired features (listed in
+no particular order):
+
+1. A coroutine task needs to be awaiter/awaitable friendly. While
+    that seems obvious, it is, e.g., possible to create an
+    `await_transform` which is deleted for awaiters.
+2. For sender algorithms it is common to adapt the results using
+    suitable algorithms. When awaiting sender in a coroutine it may
+    be annoying having to transform the result into a shape which
+    is friendly to a coroutine use. Thus, it may be reasonable to
+    support rewriting certain shapes of completion signatures into
+    something different to make the use of senders easier in a
+    coroutine task.
+3. A coroutine task needs to be sender friendly: it is expected that
+    asynchronous code is often written using coroutines awaiting
+    senders. However, it isn't as obvious as not all senders will
+    necessarily be awaitable. For example neither unifex nor stdexec
+    support senders with more than `set_value` completion.
+4. It is possibly confusing and problematic if coroutines resume on
+    a different execution context than the one they were suspended
+    on.  Senders could, however, complete on an entirely different
+    scheduler than where they started. As a result, a coroutine
+    task should be scheduler affine by default, i.e., it should
+    always resume on the same scheduler. There should probably be
+    a way to opt out of scheduler affinity when the implications are
+    well understood.
+5. When using coroutines there will probably be an allocation at
+    least for the coroutine frame. To support the use in environments
+    where memory allocations using `new`/`delete` aren't supported
+    the coroutine task should support allocations using allocators.
+6. Receivers have associated environments with attributes which are
+    generally forwarded. It should be possible to provide provide a
+    user-customisable environment from the receiver used by `co_await`
```

Comment:
I know what you want to say: we should be able to specify the amount of queries that are forwarded to awaitables within our coroutine. But I didn't understand it from the text. Maybe I'm wrong?

Reply:
OK. I'll try to clarify.

Reply:
Better